### PR TITLE
feat(tsconfig): add configuration for typescript

### PR DESCRIPTION
**Pull request by mhenkens**
Typescript `4.7.x` and angular `14`
Typescript `4.8.x` and angular `14` and `15.0.x`
Typescript `4.9.x` and angular >= `15.1.x` and < `16.x`

## PR Checklist

Please check if your PR fulfills the following requirements:

-   [x] The commit message follows our guidelines: https://github.com/NationalBankBelgium/code-style/blob/master/CONTRIBUTING.md#-commit-message-guidelines
-   [ ] Tests for the changes have been added (for bug fixes / features)
-   [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Support of typescript version for `Angular 15` 

Issue Number: N/A

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: lib/tsconfig/4.8.x/ng15/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<h1 align="center">
+   code style - TypeScript 4.8.x + Angular Angular 15
+</h1>
+
+## About
+
+This sub-package hosts the [TypeScript](https://www.typescriptlang.org/) configuration for TypeScript 4.8.x. and Angular 15
```
Comment (SuperITMan) [inline lib/tsconfig/4.8.x/ng15/README.md:7]: Only the version of typescript >= 4.8.2 are compatible with Angular 15. 
Could you adapt to `for TypeScript >= 4.8.2. and Angular 15` ?

File: lib/tsconfig/4.8.x/ng15/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    "enableLegacyTemplate": false,
+    "enableResourceInlining": false,
+    "fullTemplateTypeCheck": true,
+    "generateCodeForLibraries": true,
+    "preserveWhitespaces": false,
+    "skipMetadataEmit": false,
+    "strictMetadataEmit": false,
+    "skipTemplateCodegen": false,
+    "strictTemplates": true,
+    "strictInjectionParameters": true
```
Comment (SuperITMan) [inline lib/tsconfig/4.8.x/ng15/tsconfig.json:19]: By default, angular add the rule `"strictInputAccessModifiers": true`. 
Could you test it and adapt the configuration ?

File: lib/tsconfig/4.8.x/ng14/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    "enableLegacyTemplate": false,
+    "enableResourceInlining": false,
+    "fullTemplateTypeCheck": true,
+    "generateCodeForLibraries": true,
+    "preserveWhitespaces": false,
+    "skipMetadataEmit": false,
+    "strictMetadataEmit": false,
+    "skipTemplateCodegen": false,
+    "strictTemplates": true,
+    "strictInjectionParameters": true
```
Comment (SuperITMan) [inline lib/tsconfig/4.8.x/ng14/tsconfig.json:19]: By default, angular add the rule `"strictInputAccessModifiers": true`. 
Could you test it and adapt the configuration ?

File: lib/tsconfig/4.7.x/ng14/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    "enableLegacyTemplate": false,
+    "enableResourceInlining": false,
+    "fullTemplateTypeCheck": true,
+    "generateCodeForLibraries": true,
+    "preserveWhitespaces": false,
+    "skipMetadataEmit": false,
+    "strictMetadataEmit": false,
+    "skipTemplateCodegen": false,
+    "strictTemplates": true,
+    "strictInjectionParameters": true
```
Comment (SuperITMan) [inline lib/tsconfig/4.7.x/ng14/tsconfig.json:19]: By default, angular add the rule `"strictInputAccessModifiers": true`. 
Could you test it and adapt the configuration ?

File: lib/tsconfig/4.7.x/ng13/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    "enableLegacyTemplate": false,
+    "enableResourceInlining": false,
+    "fullTemplateTypeCheck": true,
+    "generateCodeForLibraries": true,
+    "preserveWhitespaces": false,
+    "skipMetadataEmit": false,
+    "strictMetadataEmit": false,
+    "skipTemplateCodegen": false,
+    "strictTemplates": true,
+    "strictInjectionParameters": true
```
Comment (SuperITMan) [inline lib/tsconfig/4.7.x/ng13/tsconfig.json:19]: By default, angular add the rule `"strictInputAccessModifiers": true`. 
Could you test it and adapt the configuration ?

File: lib/tsconfig/4.7.x/ng13/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  },
+  "target": "es2015"
```
Comment (SuperITMan) [inline lib/tsconfig/4.7.x/ng13/tsconfig.json:21]: Is it required to set the target to `es2015` here ?
If so, could you add a description about this in the commit message ?

File: lib/tsconfig/4.9.x/ng15/README.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
+<h1 align="center">
+   code style - TypeScript 4.9.x + Angular Angular 15.1.x || 15.2.x
+</h1>
+
+## About
+
+This sub-package hosts the [TypeScript](https://www.typescriptlang.org/) configuration for TypeScript 4.9.x. and Angular 15.1.x || 15.2.x
```
Comment (SuperITMan) [inline lib/tsconfig/4.9.x/ng15/README.md:7]: The version `4.9` can be used all the 15.x versions of Angular. Please adapt to `and Angular 15`.

**Review by SuperITMan — CHANGES_REQUESTED** [review]

File: lib/tsconfig/4.7.x/ng13/README.md
Comment (SuperITMan) [inline lib/tsconfig/4.7.x/ng13/README.md:1]: It seems that angular 13.x is not compatible with typescript 4.7.x. 
Could you remove the ng13 configuration here ?

**Review by SuperITMan — CHANGES_REQUESTED** [review]

File: lib/tsconfig/4.7.x/ng13/tsconfig.json (hunk 6 above)
Comment (mhenkens) [reply lib/tsconfig/4.7.x/ng13/tsconfig.json:21]: File remove because no support ng13 with ts4.7

File: lib/tsconfig/4.8.x/ng15/README.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
+<h1 align="center">
+   code style - TypeScript 4.8.x + Angular Angular 15
```
Comment (ageorges-nbb) [inline lib/tsconfig/4.8.x/ng15/README.md:2]:
```suggestion
   code style - TypeScript >= 4.8.2 + Angular 15
```

File: lib/tsconfig/4.8.x/ng14/README.md [hunk 9]
```diff
@@ -0,0 +1,20 @@
+<h1 align="center">
+   code style - TypeScript 4.8.x + Angular Angular 14
```
Comment (ageorges-nbb) [inline lib/tsconfig/4.8.x/ng14/README.md:2]:
```suggestion
   code style - TypeScript 4.8.x + Angular 14 
```

File: lib/tsconfig/4.9.x/ng15/README.md [hunk 10]
```diff
@@ -0,0 +1,20 @@
+<h1 align="center">
+   code style - TypeScript 4.9.x + Angular Angular 15.x
```
Comment (ageorges-nbb) [inline lib/tsconfig/4.9.x/ng15/README.md:2]:
```suggestion
   code style - TypeScript 4.9.x + Angular 15.x
```

File: lib/tsconfig/4.9.x/ng15/README.md [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+</h1>
+
+## About
+
+This sub-package hosts the [TypeScript](https://www.typescriptlang.org/) configuration for TypeScript 4.9.x. and Angular 15.x
```
Comment (ageorges-nbb) [inline lib/tsconfig/4.9.x/ng15/README.md:7]:
```suggestion
This sub-package hosts the [TypeScript](https://www.typescriptlang.org/) configuration for TypeScript 4.9.x and Angular 15.x
```

File: lib/tsconfig/4.8.x/ng15/README.md [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+</h1>
+
+## About
+
+This sub-package hosts the [TypeScript](https://www.typescriptlang.org/) configuration for TypeScript >= 4.8.2. and Angular 15
```
Comment (ageorges-nbb) [inline lib/tsconfig/4.8.x/ng15/README.md:7]:
```suggestion
This sub-package hosts the [TypeScript](https://www.typescriptlang.org/) configuration for TypeScript >= 4.8.2 and Angular 15
```

File: lib/tsconfig/4.7.x/ng14/README.md [hunk 13]
```diff
@@ -0,0 +1,20 @@
+<h1 align="center">
+   code style - TypeScript 4.7.x + Angular 14
+</h1>
+
+## About
+
+This sub-package hosts the [TypeScript](https://www.typescriptlang.org/) configuration for TypeScript 4.7.x. and Angular 14
```
Comment (ageorges-nbb) [inline lib/tsconfig/4.7.x/ng14/README.md:7]:
```suggestion
This sub-package hosts the [TypeScript](https://www.typescriptlang.org/) configuration for TypeScript 4.7.x and Angular 14
```

File: lib/tsconfig/README.md [hunk 14]
```diff
@@ -40,6 +40,14 @@ The following versions are available (based on **TypeScript** version and **Angu
 - 4.6.x - `@nationalbankbelgium/code-style/tsconfig/4.6.x`
 - 4.6.x and Angular 13 - `@nationalbankbelgium/code-style/tsconfig/4.6.x/ng13`
 - 4.6.x and Angular 14 - `@nationalbankbelgium/code-style/tsconfig/4.6.x/ng14`
+- 4.7.x - `@nationalbankbelgium/code-style/tsconfig/4.7.x`
+- 4.7.x and Angular 13 - `@nationalbankbelgium/code-style/tsconfig/4.7.x/ng13`
```
Comment (ageorges-nbb) [inline lib/tsconfig/README.md:44]: This line should be removed

**Review by ageorges-nbb — CHANGES_REQUESTED** [review]
Please also adapt the commit message:

```txt
feat(tsconfig): add configuration for typsecript

Typescript `4.7.x` and angular `14`
Typescript `4.8.x` and angular `14` and `15.0.x`
Typescript `4.9.x` and angular >= `15.1.x` < `16.x`
```

**Review by SuperITMan — APPROVED** [review]

**Pull request merged**